### ISSUE TITLE
Local server without Admin Privileges?

### ISSUE THREAD

**Issue by devyntk** [firstwiki/wiki#25]
I'm on my school provided macbook, which I sadly, without some shady dealings and exploits (that got me in trouble last year), I cannot use admin privileges on. I can use the terminal to a certain extent using a custom coded one I made (other 3rd party ones are blocked by an application per application basis). Is there any way that we can make a tutorial on how to do this without sudo? I don't have writing privileges to install `bundler`, so the built in instructions do not work. Home-brew is iffy, but I haven't tried RVM. Any help?

**Comment by virtuald** [comment]
I believe RVM is designed to solve exactly this problem.

**Comment by virtuald** [comment]
Yeah, the recommended install is "Single-User installations (recommended) - For an isolated install within a user's $HOME, not for root."

**Comment by devyntk** [comment]
After a LOT of tinkering, I finally got it to run locally. I had two main things I had to work around:
- No admin rights on my Macbook
- Custom Root CA for school monitoring

RVM recommends to use gpg to add their private key in the instructions, but gpg is not installed by default, and the only two ways to install it on mac either required home-brew (needs admin rights) or a standalone application (needs admin rights as well). I just tried to install it without gpg and it complained, but still installed fine. I then installed Ruby 2.3 easily with no fuss, but then I ran into another problem when trying to install bundler: The root CA. In the past, I've had to manually install the root CA into the java executable to make it work, but the issue was that I had no idea where openSSL stored it certificates, nor what format. It was a simple task to convert my .crt to .pem, but it was looking inside a home-brew folder for the openSSL certificates (which had only been half installed due to bad privileges earlier). That's what a utility told me, but after some searching, I found openSSL actually using `~/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/rubygems/ssl_certs`. After placing the root CA in there, I installed the gems just fine and ran it perfectly. I also had to fix the Gemfile to adjust for the SSL cert, but I just changed it to a non-ssl link instead of trying to find the cert folder.
